### ISSUE TITLE
Spurious error `'import' and 'export' may appear only with 'sourceType: module'` in <script type="module">

### ISSUE THREAD

**Issue by jameshfisher** [Munter/netlify-plugin-checklinks#286]
Thanks for this plugin! It has caught quite a few genuine errors, but here's one that I think is spurious. 

The essence of the page is:

```html
<script type="module">
  import { h, Component, render } from 'https://unpkg.com/preact?module';
  // ...
</script>
```

This gives me the error:

```
12:05:11 PM:   ✖ FAIL Failed loading inline JavaScript in foo/bar/baz.html
12:05:11 PM:   | operator: error
12:05:11 PM:   |   actual: inline JavaScript in foo/bar/baz.html: Parse error in inline JavaScript in foo/bar/baz.html
```

I'm guessing your JS parser needs to know about the `type` of the `<script>`.

**Comment by Munter** [comment]
I think you might be right on the parser error.

@papandreou this is all pretty long ago in my mind. Could you remind me what the state of JS parsing capabilities of modern modules is in assetgraph?

**Comment by papandreou** [comment]
I cannot reproduce that? https://github.com/Munter/hyperlink/commit/7ce64a99ff7e48fd476fa069cf47141bfa6273a9

Which version of hyperlink are you using? What does `npm ls assetgraph` output? (I tested with 6.2.2)

**Comment by papandreou** [comment]
I think the most likely cause is that there some other problem or syntax error in the parts of the inline script that you omitted (`// ...`), which means that we end up in the fallback to parsing it as `sourceType: 'script'` here: https://github.com/assetgraph/assetgraph/blob/19024af0e6a16bf9c5a8a86e6d40a16e405fecd1/lib/assets/JavaScript.js#L232

If that's the case, I do agree that the error message could be better. We could maybe detect errors referring to features that are only supported for modules in the fallback, and then emit the original error in that case.

**Comment by jameshfisher** [comment]
I think you're right! I came up with this minimal test case:

```
<!doctype html>
<html>
  <body>
    <script type="module">
      import { Component } from 'https://unpkg.com/preact?module';
      class Foo { state = {}; }
    </script>
  </body>
</html>
```

The `state = {};` is sort of an error on my part. What I meant to write was `class Foo { constructor() { this.state = {}; } }`. It seems what I unwittingly declared was a [field declaration](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Public_field_declarations), which works much the same way, so went unnoticed. Public field declarations are an experimental feature, but apparently have support in Chrome. I'm guessing `hyperlink`'s parser does not.

I suppose the behavior I would like is, if `hyperlink` can't parse the `module` JS, just show me the parse error, rather than catching the error and trying again as `script`.

Either way, thanks for the help!

**Comment by papandreou** [comment]
Fixed in assetgraph 6.3.0 by reporting the original error (via https://github.com/assetgraph/assetgraph/commit/918f2f9f3eca1297016e016258b273c89167a9c1) which is in range. Try updating :hugs:
